deebotozmo/cli.py

```diff
@@ -0,0 +1,542 @@
+        parser.read_file(itertools.chain(["[global]"], file), source=config_file())
+    return parser["global"]
+
+
+def write_config(config: dict) -> None:
+    """Create a new config file."""
+    os.makedirs(os.path.dirname(config_file()), exist_ok=True)
+    with open(config_file(), "w", encoding="utf-8") as file:
+        for key in config:
+            file.write(f"{key}={config[key]}\n")
+
+
+@click.group(chain=True, help="")
+@click.option("--debug/--no-debug", default=False)
+@click.option("--device", default=None, help="Select a device.")
+@click.pass_context
+def cli(
+    ctx: Union[click.Context, None] = None,
+    debug: bool = False,
+    device: Union[str, None] = None,
+) -> None:
+    """Create a click group for nesting subcommands."""
+    logging.basicConfig(format="%(name)-10s %(levelname)-8s %(message)s")
+    logging.root.setLevel(logging.DEBUG if debug else logging.WARNING)
+
+    # ensure that ctx.obj exists and is a dict (in case `cli()` is called
+    # by means other than the `if` block below)
+    if isinstance(ctx, click.Context):
+        ctx.ensure_object(dict)
+        ctx.obj["DEVICE"] = device
+
+
+@cli.command(name="createconfig", help="logs in with specified email; run this first")
+@click.option("--email", prompt="Ecovacs app email")
+@click.option("--password", prompt="Ecovacs app password", hide_input=True)
+@click.option("--country-code", prompt="your two-letter country code")
+@click.option("--continent-code", prompt="your two-letter continent code")
+@click.option("--verify-ssl", prompt="Verify SSL for API requests", default=True)
+@coro
+async def create_config(
+    email: str, password: str, country_code: str, continent_code: str, verify_ssl: str
+) -> None:
+    """Click subcommand to create a new config."""
+    if config_file_exists() and not click.confirm("overwrite existing config?"):
+        click.echo("Skipping createconfig.")
+        sys.exit(0)
+    config = {}
+    password_hash = md5(password)
+    device_id = md5(str(time.time()))
+    async with aiohttp.ClientSession() as session:
+        try:
+            EcovacsAPI(
+                session,
+                device_id,
+                email,
+                password_hash,
+                continent=continent_code,
+                country=country_code,
+                verify_ssl=verify_ssl,
+            )
+        except ValueError as error:
+            click.echo(error.args[0])
+            sys.exit(1)
+    config["email"] = email
+    config["password_hash"] = password_hash
+    config["device_id"] = device_id
+    config["country"] = country_code.lower()
+    config["continent"] = continent_code.lower()
+    config["verify_ssl"] = verify_ssl
+    write_config(config)
+    click.echo("Config saved.")
+    sys.exit(0)
+
+
+async def run_with_login(
+    device: Union[str, None],
+    cmd: Callable,
+    cmd_list: Union[list, None] = None,
+    cmd_dict: Union[dict, None] = None,
```

Comment:
```suggestion
    context: click.Context,
    cmd: Callable,
    cmd_args: Union[list, dict, None] = None,
```

Please pass the context in, so you don't need `ctx.obj["DEVICE"]` everywhere.
Also use one parameter of args. In the function that if it's a list or a dict